src/components/AddItemForm/AddItemForm.js

```diff
@@ -0,0 +1,115 @@
+import { useState } from 'react';
+import { db } from '../../lib/firebase.js';
+
+const AddItemForm = ({ listId }) => {
+  const defaultFormValues = {
+    itemName: '',
+    purchaseInterval: '7',
+  };
+
+  const [formValues, setFormValues] = useState(defaultFormValues);
+
+  // generic function updates formValues state for any of the below form inputs
+  const handleChange = (event) => {
+    const inputName = event.target.name;
+    setFormValues({ ...formValues, [inputName]: event.target.value });
+  };
+
+  // on form submit, add user input and other default values to database
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    const newItem = formValues;
+    newItem.purchaseInterval = parseInt(newItem.purchaseInterval); // convert purchaseInterval to integer
+    newItem.lastPurchaseDate = null;
+    newItem.numberOfPurchases = 0;
```

Comment:
I really like the way you set this up!

Reply:
Thanks @nick-zanetti! I know not super relevant for our project, but I like to do form values in a single state object with a single onChange if I can, because it makes it really painless if you ever need to change the form around and add/remove fields. Plus nice and concise!